### PR TITLE
cgroup: allow files containing multiple '.'

### LINKED ISSUE #2800: [Bug] Properties in cgroup2 with length > 2
# Describe the bug

In `cgroup version 2`, cgroup attribute files may have lengths greater than 2. For example, `hugetlb.1GB.max` and `memory.swap.max` are two attributes that should be able to set with `Jailer` process. However, the current `jailer` implementation assumes the length of cgroup attribute to be 2.

For the full documentation of cgroup2: [Control Group v2 - The Linux Kernel Documentation](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html).

## To Reproduce

```
jailer ... \
       --cgroup-version 2 \
       --cgroup memory.swap.max 10G
```

## Expected behaviour

It should not fail.

## Solution

The cause of the source code is in `firecracker/src/jailer/src/cgroup.rs`:

```
// Extract the controller name from the cgroup file. The cgroup file must follow
// this format: <cgroup_controller>.<cgroup_property>.
fn get_controller_from_filename(file: &str) -> Result<&str> {
    let v: Vec<&str> = file.split('.').collect();

    // Check format <cgroup_controller>.<cgroup_property>
    if v.len() != 2 {
        return Err(Error::CgroupInvalidFile(file.to_string()));
    }

    Ok(v[0])
}
```

We should change `if v.len() != 2` to `if v.len() < 2` to solve this issue.

### PR DESCRIPTION
**Pull request by luminitavoicu**
Signed-off-by: Luminita Voicu <lumivo@amazon.com>

# Reason for This PR

Fixes #2800:
Jailer's cgroup implementation considers that cgroup controllers and their associated properties follow the syntax of controller-name.property, both being separated by a dot.

However some cgroup controllers expose read/write properties that themselves contain dots that can be used to constrain the resource of a specific controller.

## Description of Changes

`[Author TODO: add description of changes.]`
- Change jailer behavior to accept properties that contain multiple '.'

- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [X] All commits in this PR are signed (`git commit -s`).
- ~[ ] The issue which led to this PR has a clear conclusion.~
- ~[ ] This PR follows the solution outlined in the related issue.~
- [X] The description of changes is clear and encompassing.
- ~[ ] Any required documentation changes (code and docs) are included in this PR.~
- ~[ ] Any newly added `unsafe` code is properly documented.~
- ~[ ] Any API changes follow the [Runbook for Firecracker API changes][2].~
- [X] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [X] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

File: tests/integration_tests/security/test_jail.py [hunk 1]
```diff
@@ -282,7 +282,7 @@ def test_cgroups(test_microvm_with_initrd, sys_setup_cgroups):
     test_microvm.jailer.numa_node = 0
     test_microvm.jailer.cgroup_ver = sys_setup_cgroups
     if test_microvm.jailer.cgroup_ver == 2:
-        test_microvm.jailer.cgroups = ['cpu.weight=2']
```
Comment (raduiliescu) [inline tests/integration_tests/security/test_jail.py:285]: This change looks unrelated.

Comment (luminitavoicu) [reply tests/integration_tests/security/test_jail.py:285]: It's not unrelated, it is meant to test the fix and ensure that controllers' properties that themselves contain dots (`weight.nice`) do not raise an error, like it did before the fix.

File: src/jailer/src/env.rs [hunk 2]
```diff
@@ -1106,6 +1106,15 @@ mod tests {
         };
         args.parse(&make_args(&invalid_cgroup_arg_vals)).unwrap();
         assert!(Env::new(&args, 0, 0).is_ok());
+
```
[the remark this answers is not in the document]
Comment (luminitavoicu) [reply src/jailer/src/env.rs:1109]: Done

**Review by alsrdn — APPROVED** [review]

File: src/jailer/src/cgroup.rs [hunk 3]
```diff
@@ -288,10 +288,13 @@ fn inherit_from_parent(path: &mut PathBuf, file_name: &str, depth: u16) -> Resul
 // Extract the controller name from the cgroup file. The cgroup file must follow
 // this format: <cgroup_controller>.<cgroup_property>.
 fn get_controller_from_filename(file: &str) -> Result<&str> {
+    if file.contains('/') {
```
Comment (alindima) [inline src/jailer/src/cgroup.rs:291]: why do we specifically perform this check?
There are other characters which are not allowed in the controller name as well, right?

File: CHANGELOG.md [hunk 4]
```diff
@@ -34,6 +34,8 @@
 - Adapt T2 and C3 CPU templates for kernel 5.10. Firecracker was not previously
   masking some CPU features of the host or emulated by KVM, introduced in more
   recent kernels: `umip`, `vmx`, `avx512_vnni`.
+- Fix jailer's croup implementation to accept properties that contain multiple
```
Comment (alindima) [inline CHANGELOG.md:37]: typo: cgroup

File: src/jailer/src/cgroup.rs (hunk 3 above)
Comment (luminitavoicu) [reply src/jailer/src/cgroup.rs]: As discussed offline, this line is not necessary. Removed.

**Review by alindima — APPROVED** [review]

**Review by alsrdn — APPROVED** [review]
